### PR TITLE
add batch prerequisites to docs

### PR DESCRIPTION
**Pull request by Riolku**
See DMOJ/judge-server#1013. There's no sense merging this before that PR.

### CODE REVIEW

**Comment by Riolku** [comment]
I want to merge DMOJ/judge-server#1019 first.

**Comment by Riolku** [comment]
The name changed in the original PR, so the commit and files have been updated.

File: docs/problem_format/problem_format.md [hunk 1]
```diff
@@ -40,6 +40,10 @@ For normal cases, the test case will contain keys `in` and `out`, mapping to the
 ## Batched Cases
 The batch will contain the keys `points` and `batched`. `batched` will map to a list of batched cases, where each case contains `in` and `out`.
 
+Optionally, the batch can contain a key `dependencies`. If specified, it should be a list of integers, indicating the one-indexed batch numbers that the batch depends on. This batch will only run if all of the dependent batches passed. The sample `init.yml` below is setup so the last batch only runs if the first two passed.
```
Comment (int-y1) [inline docs/problem_format/problem_format.md:43]:
```suggestion
Optionally, the batch can contain a key `dependencies`. If specified, it should be a list of integers, indicating the one-indexed batch numbers that the batch depends on. This batch will only run if all of the dependent batches passed. The sample `init.yml` below is set up so the last batch only runs if the first two passed.
```

**Review by int-y1 — APPROVED** [review]

**Review by Xyene — APPROVED** [review]
LGTM, thanks!

**Pull request merged**